### ISSUE TITLE
Add visualizations

### ISSUE THREAD

**Issue by tw4l** [timothyryanwalsh/METSFlask#18]
e.g. pie charts using [D3.js](https://d3js.org/)

**Comment by ablwr** [comment]
I have been playing around with this and came over to file an issue to ask if this kind of addition would be welcome --

![Screenshot from 2019-10-02 09-37-47](https://user-images.githubusercontent.com/3260492/66049022-9f786280-e4f8-11e9-9f0d-7a05cc103707.png)


I have some considerable clean-up to do; the code is super messy and styling has to be considered (colors are just hardcoded from an example instead of auto-generated for each new data point). But assuming clean-up, it seems like this would be a fine PR to make. Are there more specific things you had in mind re: visualizations/reporting, Tim?

**Comment by tw4l** [comment]
Ashley! This is so great! This is just the kind of thing I had in mind.  A great addition! I'll follow up the details in PR #22, but thank you so much!

**Comment by ablwr** [comment]
OK! Yeah it still needs a bit of cleanup, I think, for better positioning ... but glad you like it and I'll keep moving forward! I wonder what other kinds of visualizations are useful?

**Comment by tw4l** [comment]
Yay! A bit of spitballing:

* I think some sort of timeline/clustering visualization of the last modified dates could be interesting (although it's dependent on FITS, so won't be available for every file)
* Potentially variations on the format chart, so a user could select to visualize formats, format versions, or PUIDs
* What might really make these interesting/useful is if we can generalize it so that the user can select whether they want the visualizations (and table below) to be about the original files, preservation derivatives, metadata/submission documentation, or all files. This will depend on some backend changes, many of which I've already been playing with in the 'develop' branch. But for the scope of this work, it means the more we can make these generic components the better

I just merged your branch here and made the semantic changes I mentioned in the PR. Feel free to hack away on the visualizations (I'm assuming time/interest here - you've already made such great contributions here, please feel free to step away at any time!) and maybe I'll focus on the backend changes to support types of files other than original.

**Comment by tw4l** [comment]
Another idea that would also take some backend work: it could be interesting to try to visualize the relationships between the original files and their preservation derivatives - comparing formats before and after, for instance

**Comment by ablwr** [comment]
Yes! Yeah I'm still interested, even if my freetime is a bit low for this work so PRs won't be as fervent as I'd like them to be. But I gotta keep my skills up!

**Comment by tw4l** [comment]
I totally get that! Sounds good :)

**Comment by tw4l** [comment]
Hi Ashley! I did a bit more layout work on this and think it's ready to go live as-is if you're keen! Of course you're always welcome to open new PRs too :)

**Comment by ablwr** [comment]
omg thanks!! My free time has stayed low since October although I do think about this and wanting to contribute. Don't let my lack of time block you in any way.

**Comment by tw4l** [comment]
Thank you! I appreciate that you spent some of your rare free time on this, and the door will always be open if you wanna do any more! I'm going to leave this issue open as a reminder to swing back and generalize the visualization beyond formats at some point...

**Comment by ablwr** [comment]
Of course! It really wasn't much work at all -- I like to think the biggest part was motivating you to do even MORE work after me to tidy up my sloppy quickly-drafted mess. :joy:
